backend/sftp/sftp.go

```diff
@@ -427,10 +432,17 @@ func NewFs(name, root string, m configmap.Mapper) (fs.Fs, error) {
 	}
 
 	// Load key file if specified
-	if keyFile != "" {
-		key, err := ioutil.ReadFile(keyFile)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to read private key file")
+	if keyFile != "" || keyPem != "" {
+		var key []byte
+		if keyPem == "" {
+			key, err = ioutil.ReadFile(keyFile)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to read private key file")
+			}
+		} else {
+			keyPem = strings.Replace(keyPem, `\n `, "\n", -1)
```

Comment:
I note this doesn't handle `\r` but you specifically mention `\r` in the docs...

We could use [strconv.Unquote](https://golang.org/pkg/strconv/#Unquote) instead of this?

Also why does this replace `\n ` with `\n` why the trailing space?

Reply:
Yes. Thats better. Thanks!

Reply:
Since unquote expects actual quotes, i'm adding them manually to the string.  before calling unQuote but I think this is better than trying to replace strings manually.